x-pack/legacy/plugins/infra/public/pages/metrics/components/section.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent, useCallback } from 'react';
+import { EuiTitle } from '@elastic/eui';
+import { SideNavContext, SubNavItem } from '../lib/side_nav_context';
+import { LayoutProps } from '../types';
+
+type SectionProps = LayoutProps & {
+  navLabel: string;
+  sectionLabel: string;
+};
+
+export const Section: FunctionComponent<SectionProps> = ({
+  children,
+  metrics,
+  navLabel,
+  sectionLabel,
+  onChangeRangeTime,
+  isLiveStreaming,
+  stopLiveStreaming,
+}) => {
+  const { addNavItem } = React.useContext(SideNavContext);
+  const subNavItems: SubNavItem[] = [];
+
+  const childrenWithProps = React.Children.map(children, child => {
```

Comment:
Since I assume we don't want this to run on every render, should wrap this in a `useEffect` here`useEffect(...., [children, metrics])`.

Reply:
Good catch!